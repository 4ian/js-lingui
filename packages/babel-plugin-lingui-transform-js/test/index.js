import fs from 'fs'
import glob from 'glob'
import path from 'path'
import plugin from '../src/index'
import { transformFileSync } from 'babel-core'

function getTestName (testPath) {
  return path.basename(testPath)
}

describe('babel-plugin-lingui-transform-js', function () {
  glob.sync(path.join(__dirname, 'fixtures/*/')).forEach(testPath => {
    const testName = getTestName(testPath)
    // We're using relative path here to make snapshot testing work
    // across different machines.
    const actualPath = path.relative(process.cwd(), path.join(testPath, 'actual.js'))
    const expectedPath = path.join(testPath, 'expected.js')

    it(testName, () => {
      const expected = fs.existsSync(expectedPath) && fs.readFileSync(expectedPath, 'utf8').trim()

      const actual = () => transformFileSync(actualPath, {
        plugins: [
          plugin,
          'syntax-jsx',
          'transform-remove-strict-mode'
        ]
      }).code.trim()

      if (expected) {
        expect(actual()).toEqual(expected)
      } else {
        expect(actual).toThrowErrorMatchingSnapshot()
      }
    })
  })
})
