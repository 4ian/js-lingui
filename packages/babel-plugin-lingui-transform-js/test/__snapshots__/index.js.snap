// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-lingui-transform-js formats value is missing 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: The first argument of format function must be a variable."`;

exports[`babel-plugin-lingui-transform-js formats value is missing 2`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: The first argument of format function must be a variable."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal offset must be number or string, not variable 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Offset argument cannot be a variable."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural forms are missing 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Missing plural choices. At least fallback argument 'other' is required."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural forms are missing 2`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Missing plural choices. At least fallback argument 'other' is required."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural forms are missing 3`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Missing plural choices. At least fallback argument 'other' is required."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural forms cannot be variables 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Computed properties aren't allowed."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural forms missing fallback 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Missing fallback argument 'other'."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural forms missing fallback 2`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Missing fallback argument 'other'."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural forms missing fallback 3`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Missing fallback argument 'other'."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural rules must be valid 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Invalid plural rule 'three'. Must be zero, one, two, few, many, other or exact number depending on your source language ('one' and 'other' for English)."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal plural rules must be valid 2`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Invalid plural rule 'three'. Must be zero, one, two, few, many, other or exact number depending on your source language ('one' and 'other' for English)."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal value is missing 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Value argument is missing."`;

exports[`babel-plugin-lingui-transform-js validation plural/select/selectordinal value must be a variable 1`] = `"packages/babel-plugin-lingui-transform-js/test/fixtures/t/actual.js: Value must be a variable."`;
